src/test/java/mate/academy/security/AuthenticationServiceTest.java

```diff
@@ -0,0 +1,59 @@
+package mate.academy.security;
+
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceTest {
+    private static final String EMAIL = "user@gmail.com";
+    private static final String PASSWORD = "123456";
+    private AuthenticationService authenticationService;
+    private UserService userService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private User user;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        user.setEmail(EMAIL);
+        user.setPassword(PASSWORD);
+
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        authenticationService = new AuthenticationServiceImpl(
+                userService, roleService, passwordEncoder);
+    }
+
+    @Test
+    void register_ok() {
+        Mockito.when(roleService.getRoleByName("USER"))
+                .thenReturn(new Role(Role.RoleName.USER));
+        Mockito.when(userService.save(any())).thenReturn(user);
+        User actual = authenticationService.register(EMAIL, PASSWORD);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(EMAIL, actual.getEmail());
+        Assertions.assertEquals(PASSWORD, actual.getPassword());
+    }
+
+    @Test
+    void login_ok() throws AuthenticationException {
```

Comment:
Consider adding a test case for the `login` method when the login process fails due to an incorrect email or password. This will improve the test coverage. 💪